fgpyo/util/inspect.py

```diff
@@ -254,9 +315,36 @@ def dict_parse(dict_string: str) -> Dict[Any, Any]:
     return parser
 
 
+def get_fields_dict(cls: Type[DataclassesOrAttrClass]) -> Dict[str, FieldType]:
+    """
+    Get the fields dict from either a dataclasses or attr dataclass.
+
+    Combine results in case someone chooses to mix them through inheritance.
+    """
+    if not (is_dataclasses_class(cls) or is_attr_class(cls)):
+        raise TypeError("cls must a dataclasses or attr class")
+    return {
+        **(get_dataclasses_fields_dict(cls) if is_dataclasses_class(cls) else {}),
+        **(get_attr_fields_dict(cls) if is_attr_class(cls) else {}),  # type: ignore
+    }
+
+
+def get_fields(cls: Type[DataclassesOrAttrClass]) -> Tuple[FieldType, ...]:
+    if not (is_dataclasses_class(cls) or is_attr_class(cls)):
+        raise TypeError("cls must a dataclasses or attr class")
+    return (get_dataclasses_fields(cls) if is_dataclasses_class(cls) else ()) + (
+        get_attr_fields(cls) if is_attr_class(cls) else ()  # type: ignore
+    )
+
+
+AttrFromType = TypeVar("AttrFromType")
```

Comment:
Docs please.

Reply:
Added. This a typevar allowing the `attr_from` function to create either an attr class or a dataclasses class, so I added documentation there as well.

Reply:
In the future, you can document variables like:

```python
AttrFromType = TypeVar("AttrFromType")
"""TypeVar to allow attr_from to be used with either an attr class or a dataclasses class."""
```

And they will be picked up by Sphinx and compiled into rich docs (e.g. the fgpyo docs website). It might worth going back through and swapping line comments to true public variable docstrings.